httpclient/request_id_round_tripper.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"github.com/acronis/go-appkit/httpserver/middleware"
+	"net/http"
+)
+
+// RequestIDRoundTripper for X-Request-ID header to the request.
+type RequestIDRoundTripper struct {
+	Delegate http.RoundTripper
+}
+
+// NewRequestIDRoundTripper creates an HTTP transport with X-Request-ID header support.
+func NewRequestIDRoundTripper(delegate http.RoundTripper) http.RoundTripper {
+	return &RequestIDRoundTripper{
+		Delegate: delegate,
+	}
+}
+
+// RoundTrip adds X-Request-ID header to the request.
+func (rt *RequestIDRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	requestID := middleware.GetRequestIDFromContext(r.Context())
```

Comment:
Let’s add support for a `RequestIDProvider func(ctx context.Context)` string to enable customization of how request IDs are generated or retrieved. This should be included as part of the `HTTPClientOpts` struct, with a default implementation of `middleware.GetRequestIDFromContext`.

Use case: This is particularly useful for scenarios like outgoing requests from background workers, where the context is not tied to an incoming HTTP request.